R/analytics-app.R

```diff
@@ -187,24 +187,25 @@ shiny_stats_ui <- function(custom_css_path = NULL) {
   )
 }
 
-#' Stats Shiny server
-#'
-#' @param get_user function to retrieve user name
-#' @param allow_admin_rule function that asserts if user is administrator
-#' @param db_credentials data.frame with database config parameters
-#'
-#' @export
-shiny_stats_server <- function(
-  get_user, allow_admin_rule = function(session) TRUE, db_credentials
-) {
+analytics_server <- function(data_storage) {
   shiny::shinyServer(function(input, output, session) {
     session$user <- get_user(session)
 
-    data_storage <- DataStorageRSQLite$new(
-      username = session$user,
-      db_path = db_credentials$DB_NAME
-    )
+    data_storage <- data_storage
```

Comment:
This is the only way I managed to make this work. It seems like the object needs to be defined inside the server before I can pass it to a module. I guess the server can take from the environment above it but must use it before it can pass it to another inside it. 🤷🏽‍♂️